### PR TITLE
Add [with-storybook-typescript] example

### PR DESCRIPTION
**Pull request by zxzl**
Basically this PR adds Next.js 's default TypeScript setting and integrates [@storybook/preset-typescript](https://github.com/storybookjs/presets/tree/master/packages/preset-typescript) into original [with-storybook](https://github.com/vercel/next.js/tree/canary/examples/with-storybook) example. 

However, I thought it may save time for someone who are planning to use storybook with next.js and TypeScript :)

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| buildDuration | 12.9s | 13.1s | ⚠️ +205ms |
| nodeModulesSize | 67 MB | 67 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.306 | 2.229 | -0.08 |
| / avg req/sec | 1084.36 | 1121.55 | ⚠️ +37.19 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.423 | 1.472 | ⚠️ +0.05 |
| /error-in-render avg req/sec | 1756.73 | 1698.15 | -58.58 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.51 kB | 6.51 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..75a9.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.9 kB | 56.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.6 kB | 5.6 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 6.92 kB | 6.92 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| _buildManifest.js gzip | 267 B | 267 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 539 B | 539 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| index.html gzip | 955 B | 955 B | ✓ |
| link.html gzip | 961 B | 961 B | ✓ |
| withRouter.html gzip | 948 B | 948 B | ✓ |
| Overall change | 2.86 kB | 2.86 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| buildDuration | 13.9s | 13.9s | ⚠️ +17ms |
| nodeModulesSize | 67 MB | 67 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.51 kB | 6.51 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..75a9.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.9 kB | 56.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.6 kB | 5.6 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 6.92 kB | 6.92 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| _buildManifest.js gzip | 267 B | 267 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 539 B | 539 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| _error.js | 875 kB | 875 kB | ✓ |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 875 kB | 875 kB | ✓ |
| link.js | 914 kB | 914 kB | ✓ |
| routerDirect.js | 906 kB | 906 kB | ✓ |
| withRouter.js | 906 kB | 906 kB | ✓ |
| Overall change | 4.48 MB | 4.48 MB | ✓ |

</details>

</details>
Commit: a2a356670061aa8138d813352c1aea06240b7cab

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| buildDuration | 10.9s | 11s | ⚠️ +24ms |
| nodeModulesSize | 67 MB | 67 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.867 | 1.791 | -0.08 |
| / avg req/sec | 1338.83 | 1395.67 | ⚠️ +56.84 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.17 | 1.132 | -0.04 |
| /error-in-render avg req/sec | 2136.37 | 2209.19 | ⚠️ +72.82 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.51 kB | 6.51 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..75a9.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.9 kB | 56.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.6 kB | 5.6 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 6.92 kB | 6.92 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| _buildManifest.js gzip | 267 B | 267 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 539 B | 539 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| index.html gzip | 955 B | 955 B | ✓ |
| link.html gzip | 961 B | 961 B | ✓ |
| withRouter.html gzip | 948 B | 948 B | ✓ |
| Overall change | 2.86 kB | 2.86 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| buildDuration | 11.6s | 11.6s | ⚠️ +30ms |
| nodeModulesSize | 67 MB | 67 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.51 kB | 6.51 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..75a9.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.9 kB | 56.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.6 kB | 5.6 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 6.92 kB | 6.92 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| _buildManifest.js gzip | 267 B | 267 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 539 B | 539 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| _error.js | 875 kB | 875 kB | ✓ |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 875 kB | 875 kB | ✓ |
| link.js | 914 kB | 914 kB | ✓ |
| routerDirect.js | 906 kB | 906 kB | ✓ |
| withRouter.js | 906 kB | 906 kB | ✓ |
| Overall change | 4.48 MB | 4.48 MB | ✓ |

</details>

</details>
Commit: ed5af0e6eafca83589252695451df305fc2fa809

File: examples/with-storybook-typescript/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "@storybook/addon-actions": "5.3.19",
+    "@storybook/addon-links": "5.3.19",
+    "@storybook/addons": "5.3.19",
+    "@storybook/preset-typescript": "3.0.0",
+    "@storybook/react": "5.3.19",
+    "@types/node": "14.0.13",
+    "@types/react": "16.9.38",
+    "babel-loader": "^8.0.5",
+    "fork-ts-checker-webpack-plugin": "5.0.4",
+    "ts-loader": "7.0.5",
```
Comment (lfades) [inline examples/with-storybook-typescript/package.json:28]: Why are these 2 packages added?

Comment (zxzl) [reply examples/with-storybook-typescript/package.json:28]: It was required by @storybook/preset-typescript (https://github.com/storybookjs/presets/tree/master/packages/preset-typescript#installation)

**Review by lfades — APPROVED** [review]
Everything looks good, thank you!

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| buildDuration | 12.3s | 12.5s | ⚠️ +175ms |
| nodeModulesSize | 67.2 MB | 67.2 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.035 | 2.069 | ⚠️ +0.03 |
| / avg req/sec | 1228.77 | 1208.41 | -20.36 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.386 | 1.225 | -0.16 |
| /error-in-render avg req/sec | 1803.82 | 2041.2 | ⚠️ +237.38 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.5 kB | 6.5 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..834a.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.1 kB | 57.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.59 kB | 5.59 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.07 kB | 7.07 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| _buildManifest.js gzip | 267 B | 267 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 539 B | 539 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| index.html gzip | 956 B | 956 B | ✓ |
| link.html gzip | 960 B | 960 B | ✓ |
| withRouter.html gzip | 947 B | 947 B | ✓ |
| Overall change | 2.86 kB | 2.86 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| buildDuration | 13.4s | 14.2s | ⚠️ +816ms |
| nodeModulesSize | 67.2 MB | 67.2 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.5 kB | 6.5 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..834a.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.1 kB | 57.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.59 kB | 5.59 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.07 kB | 7.07 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| _buildManifest.js gzip | 267 B | 267 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 539 B | 539 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | zxzl/next.js with-storybook-typescript | Change |
| - | - | - | - |
| _error.js | 875 kB | 875 kB | ✓ |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 876 kB | 876 kB | ✓ |
| link.js | 916 kB | 916 kB | ✓ |
| routerDirect.js | 908 kB | 908 kB | ✓ |
| withRouter.js | 908 kB | 908 kB | ✓ |
| Overall change | 4.49 MB | 4.49 MB | ✓ |

</details>

</details>
Commit: 9680cd52ad89035926d12aad8d7e3fc1f74f2c55

**Pull request merged**